birdset/datamodule/embeddings_datamodule.py

```diff
@@ -0,0 +1,293 @@
+                val_data = rest_samples[num_test_samples:]
+                val_data = Dataset.from_dict({key: [sample[key] for sample in val_data] for key in val_data[0]}) #! Use first test sample as val cant be empty
+            
+            train_data = Dataset.from_dict({key: [sample[key] for sample in train_data] for key in train_data[0]})
+            test_data = Dataset.from_dict({key: [sample[key] for sample in test_data] for key in test_data[0]})
+
+            # Combine into a DatasetDict
+            dataset = DatasetDict({
+                'train': train_data,
+                'valid': val_data,
+                'test': test_data
+            })
+        else:
+            if self.val_batches == 0:
+                dataset['test'] = concatenate_datasets([dataset['valid'], dataset['test']])
+                dataset['valid'] = Dataset.from_dict({}) #! So that no split is created in the dataloader
+                
+            if self.low_train:
+                del dataset['train']
+                dataset['train'] = dataset['train_low']
+
+            del dataset['train_low']    
+                
+            
+            
+        return dataset
+
+    def _compute_embeddings(self, dataset):
+        """
+        Compute Embeddings for the entire dataset and store them in a new DatasetDict.
+        """
+        # Create a new DatasetDict to store the embeddings
+        embeddings_dataset = DatasetDict()
+        
+        #! For some reason the .map() from Datasets caused errors
+        # Iterate over each split in the dataset
+        for split in dataset.keys():
+            print(f">> Extracting Embeddings for {split} Split")
+            # Get the current split data
+            split_data = dataset[split]
+
+            # Create a list to store the embeddings for the current split
+            embeddings = []
+
+            # Iterate over each sample in the split
+            with torch.no_grad(): # No need to compute gradients
+                for sample in tqdm(split_data, total=len(split_data), desc="Extracting Embeddings"):
+                    # Get the embedding for the audio sample
+                    embedding = self._get_embedding(sample['audio'])
+                    
+                    # Add the embedding to the list
+                    sample['audio']['array'] = embedding.squeeze(0).cpu().numpy() # Move to CPU and convert to numpy
+                    embeddings.append(sample)
+
+                # Convert the list of embeddings to a tensor
+                #embeddings = torch.stack(embeddings)
+
+                # Create a new Dataset with the embeddings
+                if len(embeddings) > 0:
+                    embeddings_dataset[split] =  Dataset.from_dict({key: [sample[key] for sample in embeddings] for key in embeddings[0]})
+                else:
+                    embeddings_dataset[split] = Dataset.from_dict({'audio':[], 'labels':[]})    
```

Comment:
@XgamerTV Why do you need this? Can you not return the `dataset`? You did modify it in line 171.

Reply:
See last comment